### DISCUSSION TITLE
Code actions not return available action items.

### DISCUSSION THREAD

**Discussion by towry** [nvimdev/lspsaga.nvim#1042] [Q&A] [answered]
The volar lsp provide code action capabilities like:

```
server_capabilities = {

callHierarchyProvider = true,

codeActionProvider = {

codeActionKinds = { "", "quickfix", "refactor", "refactor.extract", "refactor.inline", "refactor.rewrite", "source", "source.fixAll", "source.organizeImports" },

resolveProvider = true

},
```

But using lspsaga's code action do not show those code actions.

If I run `lua vim.lsp.buf.code_action({ context = { only = "source" }})`, the source related action items will show correctly and works fine.

So, how to make it work with lspsaga's code action?

![截屏2023-04-26 12 43 48](https://user-images.githubusercontent.com/8279858/234472055-99a9aa06-0b2f-4491-8359-8adf973cf71a.png)
![截屏2023-04-26 12 44 10](https://user-images.githubusercontent.com/8279858/234472061-8fd1aa40-e727-42f2-8501-66c1673a28b0.png)

**Comment by glepnir** [comment]
```lua
require('lspsaga.codeaction').code_action( {context = { only = 'source'})
```

**Reply by towry** [reply]
I checked the source code, it seems not support pass context.

and I tried `require('lspsaga.codeaction'):code_action( {context = { only = 'source'})`, it doesn't work.

**Reply by glepnir** [reply]
oh sry I changed it in next version branch ..so not main branch

**Reply by glepnir** [reply, marked as answer]
this is next version code of code action. this is will work in your situation. please wait I will release this version as fast as i can ..

![image](https://user-images.githubusercontent.com/41671631/234487657-d895d864-47fe-4291-804e-90621a14aa37.png)
